app/models/recommendation.rb

```diff
@@ -25,6 +25,52 @@ def validate(record)
 
   before_validation :set_pending_status
 
+  def self.create_by_id_and_email(recommender, params)
+    recommendee_ids = params["recommendee_ids"]
+    recommendee_emails = params["recommendee_emails"]
+    item_class = Kernel.const_get(params['item_type'])
+    item = item_class.find(params['item_id'])
+
+    new_recommendations = []
+
+    unless recommendee_ids.nil?
```

Comment:
I haven't thought of a better way to do this, but this method looks verryyyy long to me. Sure you can't split it up into simpler chunks?